anchore_manager/util/db.py

```diff
@@ -0,0 +1,217 @@
+
+        try:
+            if localconfig and "do_db_post_actions" in dir(upgrade_module):
+                logger.info("DB post actions: running...")
+                upgrade_module.do_db_post_actions(localconfig=localconfig)
+        except Exception as err:
+            raise err
+
+    return code_versions, db_versions
+
+
+def db_preflight(db_params: dict = None, db_retries=3):
+    """
+    Check the configuration and verify the db is running
+
+    :param config:
+    :param db_connect:
+    :param db_use_ssl:
+    :param db_timeout:
+    :param db_connect_timeout:
+    :param db_retries:
+    :return:
+    """
+    # do some DB connection/pre-checks here
+    connected_db_params = connect_database(db_params, db_retries=db_retries)
+    return connected_db_params
+
+
+def needs_upgrade(code_versions, db_versions):
+    """
+    Check if an upgrade is needed
+
+    :param code_versions:
+    :param db_versions:
+    :return: None if no db upgrade needed, or tuple of (code db version (str), running db version (str)) if an upgrade is needed
+    """
+
+    code_db_version = code_versions.get('db_version', None)
+    running_db_version = db_versions.get('db_version', None)
+
+    if not code_db_version or not running_db_version:
+        raise Exception("cannot get version information (code_db_version={} running_db_version={})".format(code_db_version, running_db_version))
+
+    if code_db_version == running_db_version:
+        return None
+    else:
+        return code_db_version, running_db_version
+
+
+def load_upgrade_module(module_name: str):
+    """
+    Load the named module, verifying it, and return it loaded
+
+    :param module_name:
+    :return:
+    """
+    try:
+        logger.info("Loading DB upgrade routines from module {}".format(module_name))
+        return importlib.import_module(module_name)
+    except Exception as err:
+        raise Exception("Input module (" + str(module_name) + ") cannot be found/imported - exception: " + str(err))
+
+
+def upgrade_db(code_versions: dict, db_versions: dict, upgrade_module):
+    """
+    Run the upgrade process for the given module. Raises exception on errors, caller must handle end exit cleanly.
+    Expects that the db has been initialized already via call to init_database() or similar
+
+    :param code_versions: dict with versions for the code found installed
+    :param db_versions: dict with versions for the versions found stored in the db (typically returned from init_database() call
+    :param upgrade_module:
+    :return: running db_version after upgrade
+    """
+    # Load the module for upgrade (provides the upgrade routines etc
+    module = upgrade_module
+
+    versions_tuple = needs_upgrade(code_versions, db_versions)
+    if versions_tuple:
+        code_db_version = versions_tuple[0]
+        running_db_version = versions_tuple[1]
+        logger.info("Detected anchore-engine version {}, running DB version {}.".format(code_db_version, running_db_version))
```

Comment:
lets avoid `.format()` in logging calls
```suggestion
        logger.info("Detected anchore-engine version %s, running DB version %s.", code_db_version, running_db_version)
```